js/integration-tests/solidity-tests/test/fuzz.ts

```diff
@@ -486,4 +492,125 @@ describe("Fuzz and invariant testing", function () {
       }
     }
   });
+
+  it("FuzzFunctionOverrides", async function () {
+    const artifact = testContext.matchingTest("FuzzConfigOverrideTest")[0];
+
+    const fuzzConfig = {
+      runs: 100,
+      maxTestRejects: 0,
+    };
+
+    const testFunctionOverrides = [
+      {
+        identifier: {
+          contractArtifact: artifact,
+          functionSelector: "0x6d301800", // testFuzz_OverrideRuns(uint256)
+        },
+        config: {
+          fuzz: {
+            runs: 10,
+          },
+        },
+      },
+      {
+        identifier: {
+          contractArtifact: artifact,
+          functionSelector: "0x1a2888a7", // testFuzz_OverrideTimeoutAndRejects(uint256)
+        },
+        config: {
+          fuzz: {
+            runs: 256,
+            maxTestRejects: 50000,
+            timeout: {
+              time: 1,
+            },
+          },
+        },
+      },
+      {
+        identifier: {
+          contractArtifact: artifact,
+          functionSelector: "0xff055599", // testFuzz_NoOverrideTimeout(uint256)
+        },
+        config: {
+          fuzz: {
+            maxTestRejects: 5000,
+          },
+        },
+      },
+    ];
+
+    const result = await testContext.runTestsWithStats(
+      "FuzzConfigOverrideTest",
+      {
+        fuzz: fuzzConfig,
+        testFunctionOverrides: testFunctionOverrides,
+      }
+    );
+
+    assert.equal(result.failedTests, 2);
+    assert.equal(result.totalTests, 5);
+
+    assert.equal(
+      result.stackTraces.get("testFuzz_NoOverrideRejects(uint256)")?.reason,
+      "`vm.assume` rejected too many inputs (0 allowed)"
+    );
+
+    assert.equal(
+      result.stackTraces.get("testFuzz_NoOverrideTimeout(uint256)")?.reason,
+      "`vm.assume` rejected too many inputs (5000 allowed)"
+    );
+
+    const suite_result = result.suiteResults[0];
+    for (const test_result of suite_result.testResults) {
+      if (test_result.name === "testFuzz_OverrideRuns(uint256)") {
+        const fuzzKind = test_result.kind as FuzzTestKind;
+        assert.equal(fuzzKind.runs, 10n); // Overridden
```

Comment:
Suggestion(non-blocking): Can these numeric literals be defined as constants and shared between the setup and test validation?